### PR TITLE
add create namespace command

### PR DESCRIPTION
**Pull request by mattkim**
## What was changed

add create namespace command

## Why?

allow users to create a namespace using `tcld`

## Checklist

- feature parity with create namespace in `cloud.temporal.io`
- supports creating namespace with:
  - name
  - region
  - retention (in days)
  - ca-certificate
  - certificate-filters
  - search attributes
  - user namespace permissions

1. How was this tested:

unit tests

### CODE REVIEW

File: app/namespace.go [hunk 1]
```diff
@@ -184,6 +272,129 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
...
+							Required: true,
+						},
+						&cli.IntFlag{
+							Name:    RetentionDaysFlagName,
+							Usage:   "The retention of the namespace in days",
+							Aliases: []string{"rd"},
+							Value:   30,
+						},
+						&cli.StringFlag{
+							Name:    CaCertificateFlagName,
```
Comment (anekkanti) [inline app/namespace.go:296]: There is an existing flag that we could reuse: https://github.com/temporalio/tcld/blob/main/app/namespace.go#L26

File: app/namespace.go [hunk 2]
```diff
@@ -5,24 +5,45 @@ import (
...
 )
 
 var (
+	namespaceRegionFlag = &cli.StringFlag{
+		Name:     namespaceRegionFlagName,
+		Usage:    "Create namespace in this region",
+		Aliases:  []string{"re"},
+		Required: true,
+	}
+	namespaceEnvironmentFlag = &cli.StringFlag{
```
Comment (mastermanu) [inline app/namespace.go:40]: what's the point of even adding this if its hidden right now? I believe server will auto-assign the value if its not set anyway, right?

File: app/namespace.go [hunk 3]
```diff
@@ -33,23 +54,30 @@ var (
...
 	caCertificateFingerprintFlag = &cli.StringFlag{
 		Name:    caCertificateFingerprintFlagName,
 		Usage:   "The fingerprint of to the ca certificate",
 		Aliases: []string{"fp"},
 	}
+	searchAttributeFlag = &cli.StringSliceFlag{
+		Name:     searchAttributeFlagName,
+		Usage:    fmt.Sprintf("Flag can be used multiple times; value must be \"name=type\"; valid types are: %v", getSearchAttributeTypes()),
+		Aliases:  []string{"sa"},
+		Required: true,
```
Comment (mastermanu) [inline app/namespace.go:66]: why is this required?

Comment (mastermanu) [reply app/namespace.go:66]: (or where is this getting used? maybe I have missed it)

File: app/namespace.go [hunk 4]
```diff
@@ -145,6 +186,33 @@ func (c *NamespaceClient) parseExistingCerts(ctx *cli.Context) (namespace *names
 	return n, existingCerts, nil
 }
 
+func (c *NamespaceClient) toUserNamespacePermissions(userPermissionsInput map[string]string) ([]*auth.UserNamespacePermissions, error) {
+	var res []*auth.UserNamespacePermissions
+	for email, actionGroup := range userPermissionsInput {
+		u, err := c.authClient.GetUser(c.ctx, &authservice.GetUserRequest{
```
Comment (mastermanu) [inline app/namespace.go:192]: is this potentially going to cause any rate limiting by the API?

File: app/namespace.go [hunk 5]
```diff
@@ -145,6 +186,33 @@ func (c *NamespaceClient) parseExistingCerts(ctx *cli.Context) (namespace *names
...
+			UserEmail: email,
+		})
+		if err != nil {
+			return nil, err
+		}
+		if len(u.GetUser().GetId()) == 0 {
+			return nil, fmt.Errorf("user not found for: %s", email)
```
Comment (mastermanu) [inline app/namespace.go:199]: [Nit] consider using multierror so the user can get a list of all the issues that are wrong instead of having to do one at a time

File: app/namespace.go [hunk 6]
```diff
@@ -184,6 +272,125 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
...
+						}
+						if len(certFilterBytes) > 0 {
+							newFilters, err := parseCertificateFilters(certFilterBytes)
+							if err != nil {
+								return err
+							}
+							n.Spec.CertificateFilters = append(n.Spec.CertificateFilters, newFilters.toSpec()...)
+						}
+
+						// search attribute (optional)
```
Comment (mastermanu) [inline app/namespace.go:372]: [Nit] //search attributes

File: app/namespace.go (hunk 3 above)
Comment (mattkim) [reply app/namespace.go:67]: this is used by the update namespace command where the flag is required.

File: app/namespace.go [hunk 7]
```diff
@@ -184,6 +272,129 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
...
 			Subcommands: []*cli.Command{
+				{
+					Name:    "create",
+					Usage:   "Create a temporal namespace",
+					Aliases: []string{"c"},
+					Flags: []cli.Flag{
+						RequestIDFlag,
+						namespaceRegionFlag,
+						namespaceEnvironmentFlag,
+						&cli.StringFlag{
```
[the remark this answers is not in the document]
Comment (mattkim) [reply app/namespace.go:289]: wanted to set a different default value--open question on if we should default to 30?

File: app/namespace.go (hunk 1 above)
Comment (mattkim) [reply app/namespace.go]: will reuse this one.

File: app/namespace.go (hunk 2 above)
Comment (mattkim) [reply app/namespace.go:41]: removed the flag--you are right it seems to work without it

File: app/namespace.go (hunk 4 above)
Comment (mattkim) [reply app/namespace.go:174]: it could but what options do we have here? some kind of artificial sleep?

Comment (mattkim) [reply app/namespace.go:185]: agreed offline to move this fetching of users to the server side as part of another pr.

File: app/namespace.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (mattkim) [reply app/namespace.go:278]: Actually it should be 30 days--reverting this back.

**Review by mastermanu — APPROVED** [review]

**Review by anekkanti — APPROVED** [review]
